### PR TITLE
Addressed comments from #529.

### PR DESCRIPTION
**Pull request by eaplatanios**
This PR addresses @rxwei's comments from #529.

### CODE REVIEW

File: Sources/TensorFlow/Core/Serialization.swift [hunk 1]
```diff
@@ -67,16 +64,30 @@ public class TensorFlowCheckpointReader {
         return TensorShape([Int64](dimensionsBufferPointer).map(Int.init))
     }
 
-    /// Returns the data type of the tensor with the provided name stored in the checkpoint.
-    @inlinable
-    public func dataType(ofTensorNamed name: String) -> TensorDataType {
-        TensorDataType(TF_CheckpointReaderGetVariableDataType(handle, name))
+    /// Returns the type of the tensor with the provided name stored in the checkpoint.
+    public func typeOfTensor(named name: String) -> Any.Type {
```
Comment (rxwei) [inline Sources/TensorFlow/Core/Serialization.swift]:
```suggestion
    public func scalarTypeOfTensor(named name: String) -> Any.Type {
```

File: Sources/TensorFlow/Core/Serialization.swift [hunk 2]
```diff
@@ -20,13 +20,13 @@ public class TensorFlowCheckpointReader {
     @usableFromInline internal let status: OpaquePointer
```
Comment (rxwei) [inline Sources/TensorFlow/Core/Serialization.swift]: `@usableFromInline` can be all removed.

Comment (eaplatanios) [reply Sources/TensorFlow/Core/Serialization.swift]: Done.

File: Sources/TensorFlow/Core/Serialization.swift (hunk 1 above)
Comment (eaplatanios) [reply Sources/TensorFlow/Core/Serialization.swift]: Done.

Comment (rxwei) [reply Sources/TensorFlow/Core/Serialization.swift]: Oh actually you can make it more specific: return `TensorFlowScalar.self`.

Comment (eaplatanios) [reply Sources/TensorFlow/Core/Serialization.swift]: That does not work because of the `String` type.

**Pull request merged**